File: new.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

''' create a file
example:
python Scripts/new.py -f hello/readme.md  # even folder hello not exists
'''

import argparse
import pathlib

heads = {'.py':'#!/usr/bin/env python3'}

parser = argparse.ArgumentParser(description='Create a file')
parser.add_argument('-f', action='store', dest='filename', default='script.txt', metavar='FILE', type=pathlib.Path)
parser.add_argument('-s', nargs='+', dest='content', action='store', default='', metavar='CONTENT')
parser.add_argument('-l', action='store', dest='head', metavar='HEAD')

args = parser.parse_args()

p = args.filename
if args.head:
    s = '%s\n\n%s'%(args.head, '\n'.join(args.content))
else:
    head = heads.get(p.suffix, '')
    s = '%s\n\n%s'%(head, '\n'.join(args.content))
try:
    p.touch()
except FileNotFoundError:
    p.parent.mkdir(parents=True)
    p.touch()
p.write_text(s)

File: new.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

=begin 
create a file
example:
python Scripts/new.py -f hello/readme.md  # even folder hello not exists
=end

require 'Thor'

$heads = {'py':'#!/usr/bin/env python3', 'rb':'#!/usr/bin/env ruby'}

class NewCLI < Thor
    # ruby new.rb new --item Python
    desc "New File", "Ruby cli for creating files"
    option :path, :aliases => :p, :default => 'test.txt'
    option :string, :aliases => :s, :default => ''
    option :file, :aliases => :f, :default => ''

    def new     
        if options[:string]
            content = options[:string]
        elsif options[:file]
            file = File.new(options[:file], "r")
            File.open("filename", "mode") do |file|
               content = file.sysread()
            end
        end

        File.open(options[:path], "w") do |file|
            suffix = options[:path].split('.')[-1]
            if $heads[suffix.to_sym]
                content = $heads[suffix.to_sym] + '/n/n' + content
            end
            file.syswrite(content)
        end
    end
end
 
NewCLI.start(ARGV)
